Hw2.2.2

leushkinml opened the pull request:
Решения по ДЗ 1 закомментировал.

leushkinml committed:
ДЗ 2: 2-е задание выполнено

leushkinml committed:
ДЗ 3: 2-е задание выполнено

leushkinml committed:
ДЗ 4: 3-е задание выполнено

leushkinml committed:
ДЗ 4: 3-е задание выполнено

leushkinml committed:
ООП Полиморфизм:
ДЗ 1: 1-е задание выполнено

leushkinml committed:
ООП Полиморфизм:
ДЗ 1: 2-е задание выполнено

leushkinml committed:
ООП Полиморфизм:
ДЗ 2: 1-е задание выполнено

leushkinml committed:
ООП Полиморфизм:
ДЗ 2: 1-е задание выполнено

dmitry-bizin commented on src/Transport.java:
```diff
@@ -0,0 +1,79 @@
+    }
+
+    public void setColor(String color) {
+        if (color == null || color.isEmpty() || color.isBlank()) {
```
isEmpty лишняя проверка, isBlank это проверяет тоже

dmitry-bizin commented on src/Car.java:
```diff
@@ -0,0 +1,257 @@
+        } else {
+            this.bodyType = bodyType;
+        }
+        if (registrationNumber == null || registrationNumber.isEmpty() || registrationNumber.isBlank()) {
```
isEmpty лишняя проверка, isBlank это проверяет тоже

dmitry-bizin commented on src/Car.java:
```diff
@@ -0,0 +1,257 @@
+        } else {
+            this.transmission = transmission;
+        }
+        if (bodyType == null || bodyType.isEmpty() || bodyType.isBlank()) {
```
isEmpty лишняя проверка, isBlank это проверяет тоже

dmitry-bizin commented on src/Car.java:
```diff
@@ -0,0 +1,257 @@
+
+        this.engineVolume = engineVolume;
+
+        if (transmission == null || transmission.isEmpty() || transmission.isBlank()) {
```
isEmpty лишняя проверка, isBlank это проверяет тоже

dmitry-bizin commented on src/Car.java:
```diff
@@ -0,0 +1,257 @@
+        return transmission;
+    }
+    public void setTransmission(String transmission) {
+        if (transmission == null || transmission.isEmpty() || transmission.isBlank()) {
```
isEmpty лишняя проверка, isBlank это проверяет тоже

dmitry-bizin commented on src/Car.java:
```diff
@@ -0,0 +1,257 @@
+               int numberOfSeats, boolean tiresType, Key key, Insurance insurance) {
+
+        super(brand, model, productionYear, productionCountry, color);
+
```
в конструкторах лучше использовать сеттеры, чтобы не дублировать проверки - все проверки в сеттерах

dmitry-bizin commented on src/Car.java:
```diff
@@ -0,0 +1,257 @@
+    }
+
+    public static class Insurance {
+        private final LocalDate durationOfInsurance;
```
хм, это я так понял дата окончания? тогда почему она у тебя ставится по дефолту как 1 января 0000 года?) может, 365 дней от текущего поставить? LocalDate.now().plusYear(1)

dmitry-bizin commented on src/Train.java:
```diff
@@ -0,0 +1,143 @@
+public class Train extends Transport {
+
+    private double tripPrice;
+    private Time travelTime;
```
не, тут лучше использовать просто int - например, кол-во минут поездки. Или Duration, если ты хочешь что-то типа Time. Но Time не для этого

dmitry-bizin commented on src/Train.java:
```diff
@@ -0,0 +1,143 @@
+            this.travelTime = travelTime;
+        }
+
+        if (departureStation == null || departureStation.isEmpty() || departureStation.isBlank()) {
```
isEmpty лишняя проверка, isBlank это проверяет тоже

dmitry-bizin commented on src/Train.java:
```diff
@@ -0,0 +1,143 @@
+            this.departureStation = departureStation;
+        }
+
+        if (endingStation == null || endingStation.isEmpty() || endingStation.isBlank()) {
```
isEmpty лишняя проверка, isBlank это проверяет тоже

dmitry-bizin commented on src/Train.java:
```diff
@@ -0,0 +1,143 @@
+    }
+
+    public void setDepartureStation(String departureStation) {
+        if (departureStation == null || departureStation.isEmpty() || departureStation.isBlank()) {
```
isEmpty лишняя проверка, isBlank это проверяет тоже

dmitry-bizin commented on src/Train.java:
```diff
@@ -0,0 +1,143 @@
+    }
+
+    public void setEndingStation(String endingStation) {
+        if (endingStation == null || endingStation.isEmpty() || endingStation.isBlank()) {
```
isEmpty лишняя проверка, isBlank это проверяет тоже

dmitry-bizin commented on src/Train.java:
```diff
@@ -0,0 +1,143 @@
+    {
+        super(brand, model, productionYear, productionCountry, maxSpeed);
+
+        if (tripPrice < 0) {
```
здесь лучше вызывать сеттеры, чтобы не дублировать проверки

dmitry-bizin commented on src/Transport.java:
```diff
@@ -0,0 +1,79 @@
+        this.color = color;
+    }
+
+    public void refill() {
```
этот метод должен быть абстрактный (см. задание)

leushkinml committed:
ООП Полиморфизм:
ДЗ 2: 1-е задание выполнено

leushkinml committed:
ООП Полиморфизм:
ДЗ 2: 1-е задание выполнено

leushkinml committed:
ООП Полиморфизм:
ДЗ 2: 1-е задание выполнено

leushkinml committed:
ООП Полиморфизм:
ДЗ 2: 1-е задание выполнено

leushkinml committed:
ООП Полиморфизм generic:
ДЗ 1: 1-е задание выполнено

leushkinml committed:
ООП Полиморфизм generic:
ДЗ 2: 2-е задание выполнено

leushkinml committed:
ООП Полиморфизм generic:
ДЗ 3: 3-е задание выполнено

leushkinml committed:
ООП Полиморфизм generic:
ДЗ 4: 4-е задание выполнено

leushkinml committed:
ООП Полиморфизм generic:
ДЗ 4: 4-е задание выполнено

mochaches commented on src/Bus.java:
```diff
@@ -0,0 +1,74 @@
+    }
+
+    @Override
+    public void startMoving() {
```
пустые методы очень сбивают с мысли. Кажется он есть, но он ничего не делает. Так делать не стоит. Если методы не нужны, не стоит наследоваться. Если наследуешься - реализуй методы

mochaches commented on src/Bus.java:
```diff
@@ -0,0 +1,74 @@
+
+    @Override
+    public void isPitStop(boolean pitStop) {
+        if (pitStop == true) {
```
pitStop уже true\false 
Нет смысла сравнивать true==true или true==false. Все равно получится тот же true или false.

mochaches commented on src/Car.java:
```diff
@@ -0,0 +1,291 @@
+
+        @Override
+        public void startMoving() {
+        }
```
аналогично

mochaches commented on src/Main.java:
```diff
@@ -1,5 +1,186 @@
+import java.sql.Time;
```
лишние импорты

mochaches commented on src/DriverB.java:
```diff
@@ -0,0 +1,33 @@
+public class DriverB extends Drivers  {
+
+
+    public static final String DL_CATEGORY = "B";
```
для чего константа?

mochaches commented on src/DriverC.java:
```diff
@@ -0,0 +1,23 @@
+public class DriverC extends Drivers {
+
+    public static final String DL_CATEGORY = "C";
```
для чего константа?

mochaches commented on src/DriverD.java:
```diff
@@ -0,0 +1,24 @@
+public class DriverD extends Drivers {
+
+
+    public static final String DL_CATEGORY = "D";
```
для чего константа?

leushkinml commented:
Лишние троки убрал

leushkinml committed:
ООП Полиморфизм generic:
ДЗ 4: 4-е задание выполнено

leushkinml committed:
Enum:
ДЗ 1: 1-е задание выполнено

leushkinml committed:
Enum:
ДЗ 2: 1-е задание выполнено

leushkinml committed:
Исключения:
ДЗ 2: 2-е задание выполнено

leushkinml committed:
Коллекции:
ДЗ 1: 1-е задание выполнено
ДЗ 2: 1-е задание выполнено

leushkinml committed:
Коллекции:
ДЗ 1: 1-е задание выполнено
ДЗ 2: 1-е задание выполнено
Хеш-функции:
ДЗ 3: 1-е задание выполнено
